go/sync2/consolidator.go

```diff
@@ -117,13 +117,31 @@ func (rs *pendingResult) SetResult(res *sqltypes.Result) {
 	rs.result = res
 }
 
+
 // Wait waits for the original query to complete execution. Wait should
 // be invoked for duplicate queries.
-func (rs *pendingResult) Wait() {
+func (rs *pendingResult) Wait(ctx context.Context) error {
 	rs.consolidator.Record(rs.query)
-	rs.executing.RLock()
+	acquired := make(chan struct{})
+	go func() {
+		rs.executing.RLock()
```

Comment:
How does this stop the this new goroutine from leaking? We'd now wait indefinitely on this new lock statement here? 

Have you validated that this solves the problem and that you don't see the lake? If so, how? We also need a test for this to validate it works.